### PR TITLE
Add service offering icons

### PR DESCRIPTION
**Pull request by Ladas**
Depends on:
- [x] https://github.com/ManageIQ/topological_inventory-core/pull/91

### CODE REVIEW

**Comment by Ladas** [comment]
cc @agrare

File: app/models/service_offering.rb [hunk 1]
```diff
@@ -13,4 +14,8 @@ class ServiceOffering < ApplicationRecord
 
   acts_as_tenant(:tenant)
   acts_as_taggable_on
+
+  def icon_data
+    service_offering_icon&.data
+  end
```
Comment (agrare) [inline app/models/service_offering.rb]: We're trying to stay away from having methods in the models as much as possible to keep them clean, it is just as easy for the caller to do this.

File: app/models/service_offering_icon.rb [hunk 2]
```diff
@@ -0,0 +1,8 @@
+class ServiceOfferingIcon < ApplicationRecord
+  belongs_to :tenant
```
Comment (agrare) [inline app/models/service_offering_icon.rb]: Can you add the other side of this association to `Tenant`?

File: app/models/service_offering.rb (hunk 1 above)
Comment (Ladas) [reply app/models/service_offering.rb]: right, so this is similar to tags, so faking an attribute, so we can use it in API response

alternative here is to just expose service_offering_icon_id and service_offering_icons collection?

Comment (agrare) [reply app/models/service_offering.rb]: Ah hm, how about we start with the API matching the modeling then change it if we need to?

Comment (Ladas) [reply app/models/service_offering.rb]: ok (my thinking was starting with what is most convenient for the user)

File: db/migrate/20190118110059_add_service_offering_icons.rb [hunk 3]
```diff
@@ -0,0 +1,16 @@
+class AddServiceOfferingIcons < ActiveRecord::Migration[5.2]
+  def change
+    create_table "service_offering_icons", :id => :serial, :force => :cascade do |t|
+      t.references :tenant, :type => :bigint, :index => true, :null => false, :foreign_key => {:on_delete => :cascade}
+      t.references :source, :type => :bigint, :index => false, :null => false, :foreign_key => {:on_delete => :cascade}
```
Comment (agrare) [inline db/migrate/20190118110059_add_service_offering_icons.rb]: Why not have an index on this one?

Comment (Ladas) [reply db/migrate/20190118110059_add_service_offering_icons.rb]: yeah, adding specs for these. Source is already covered by the unique index

Comment (agrare) [reply db/migrate/20190118110059_add_service_offering_icons.rb]: Ah okay

**Pull request merged**